packages/x-tree-view/src/internals/plugins/useTreeViewFocus/useTreeViewFocus.ts

```diff
@@ -1,67 +1,79 @@
 import * as React from 'react';
 import useEventCallback from '@mui/utils/useEventCallback';
+import useEnhancedEffect from '@mui/utils/useEnhancedEffect';
 import { EventHandlers } from '@mui/utils';
-import ownerDocument from '@mui/utils/ownerDocument';
-import { TreeViewPlugin, TreeViewUsedInstance } from '../../models';
+import { TreeViewPlugin } from '../../models';
 import { UseTreeViewFocusSignature } from './useTreeViewFocus.types';
 import { useInstanceEventHandler } from '../../hooks/useInstanceEventHandler';
-import { getActiveElement } from '../../utils/utils';
 import { getFirstNavigableItem } from '../../utils/tree';
 import { TreeViewCancellableEvent } from '../../../models';
 import { convertSelectedItemsToArray } from '../useTreeViewSelection/useTreeViewSelection.utils';
-
-const useDefaultFocusableItemId = (
-  instance: TreeViewUsedInstance<UseTreeViewFocusSignature>,
-  selectedItems: string | string[] | null,
-): string => {
-  let tabbableItemId = convertSelectedItemsToArray(selectedItems).find((itemId) => {
-    if (!instance.isItemNavigable(itemId)) {
-      return false;
-    }
-
-    const itemMeta = instance.getItemMeta(itemId);
-    return itemMeta && (itemMeta.parentId == null || instance.isItemExpanded(itemMeta.parentId));
-  });
-
-  if (tabbableItemId == null) {
-    tabbableItemId = getFirstNavigableItem(instance);
-  }
-
-  return tabbableItemId;
-};
+import {
+  selectorDefaultFocusableItemId,
+  selectorFocusedItemId,
+} from './useTreeViewFocus.selectors';
+import { selectorIsItemExpanded } from '../useTreeViewExpansion/useTreeViewExpansion.selectors';
+import {
+  selectorCanItemBeFocused,
+  selectorItemMeta,
+} from '../useTreeViewItems/useTreeViewItems.selectors';
 
 export const useTreeViewFocus: TreeViewPlugin<UseTreeViewFocusSignature> = ({
   instance,
   params,
-  state,
-  setState,
+  store,
   models,
-  rootRef,
 }) => {
-  const defaultFocusableItemId = useDefaultFocusableItemId(instance, models.selectedItems.value);
-
-  const setFocusedItemId = useEventCallback((itemId: React.SetStateAction<string | null>) => {
-    const cleanItemId = typeof itemId === 'function' ? itemId(state.focusedItemId) : itemId;
-    if (state.focusedItemId !== cleanItemId) {
-      setState((prevState) => ({ ...prevState, focusedItemId: cleanItemId }));
+  useEnhancedEffect(() => {
```

Comment:
That's probably just because I lack of knowledge about the tree view. But I don't get why this `useEnhancedEffect` effect got added since I don't see any effects in the `useTreeViewFocus` before this commit

Reply:
Before this PR, we were computing `defaultFocusableItemId` in the render and then the item would access it using `instance.canItemBeTabbed`.

We can no longer use `instance` method like this AND we don't want to re-render every item everytime this variable changes (so we don't want to pass `defaultFocusableItemId` to the context).

The solution here is to move `defaultFocusableItemId` to the state and access it in the item using a selector that returns a boolean (similar to the old `instance.canItemBeTabbed`).

I added the state update in an effect but I should be able to just run it in the render to avoid rendering cascade :+1: